target/linux/ath79/generic/base-files/etc/board.d/02_network

```diff
@@ -48,7 +48,8 @@ ath79_setup_interfaces()
 	ubnt,unifiac-mesh|\
 	ubnt,unifi|\
 	wd,mynet-wifi-rangeextender|\
-	winchannel,wb2000)
+	winchannel,wb2000|\
+	zsun,sd100)
```

Comment:
Well, I thought there is no network interface ...?

Reply:
Added an empty case statement for Zsun, however I still have the eth0 added to /etc/config/network

Reply:
I fear this will require some research. I have not encountered a WiFi-only device so far on ath79 myself, I do not know what config_generate does in this case.

Reply:
> I fear this will require some research.

The issue lies within with the package ```base-files``` on the script ```files/etc/board.d/99-default_network``` (line 12):
```
ucidef_set_interface_lan 'eth0'
```

However to fix this would require some refactoring of these initialization scripts (they currently require an interface to initialize the 'lan' network node), so I guess for Zsun-SD100 I'll just leave the empty case statement until the day this refactoring is done.

Reply:
Well, this refactoring will be one of the prerequisites to make OpenWrt ready for Wifi-only devices. Just leaving the work for someone else without even having tried to think about it is a little too easy.

However, this not the primary obstacle, but one you will have to address after the boot/partitioning and the default Wifi have been dealt with.

Reply:
Despite, I really don't like it when submitters just close a discussion as resolved because they want to hide a shortcoming of their submission. I've reopened this already once, doing it a second time now, and I won't do a third.

Reply:
I've submitted a [pull request](https://github.com/openwrt/openwrt/pull/2571) for the **base-files** package with the required changes for eth0 to only be added if it's present on the device.

Reply:
@adrianschmutzler 
Would you agree to mark this as resolved since:
- This has ZERO impact on the device - it just adds the non-existent interface eth0 to the lan network;
- This is not an issue of this patch but rather the **base-files** package;
- I have already submitted a [pull request](https://github.com/openwrt/openwrt/pull/2571) to address this issue on that package.

Reply:
Well, adding a non-existent interface is not "ZERO", though it might be not too important.
I do not see what we gain from marking this as resolved until the other pull request is dealt with. We would just hide something we will stumble over later.